types/aqb/index.d.ts

```diff
@@ -0,0 +1,1773 @@
+     * If the value is already an  Number or  Integer, its own value will be wrapped instead.
+     *
+     * @param value
+     */
+    num (value: any) :  NumberLiteral;
+    /**
+     * Integer
+     *
+     * Wraps the given value as an  Integer literal.
+     *
+     * qb.int(value)
+     *
+     * If the value is not a JavaScript Number, it will be converted first.
+     * If the value does not represent a finite integer, an Error will be thrown.
+     * If the value is already an  Number or  Integer, its own value will be wrapped instead.
+     *
+     * @param value
+     */
+    int (value: any) :  IntegerLiteral;
+    /**
+     * String
+     *
+     * Wraps the given value as an  String literal.
+     *
+     * qb.str(value)
+     *
+     * If the value is not a JavaScript String, it will be converted first.
+     * If the value is a quoted string, it will be treated as a string literal.
+     * If the value is an object with a toAQL method, the result of calling that method will be wrapped instead.
+     *
+     * Examples
+     *
+     * 23 => "23"
+     *
+     * "some string" => "some string"
+     *
+     * '"some string"' => "\"some string\""
+     *
+     *
+     * @param value
+     */
+    str (value: any) :  StringLiteral;
+    /**
+     * List
+     *
+     * Wraps the given value as an  List (Array) literal.
+     *
+     * qb.list(value)
+     *
+     * If the value is not a JavaScript Array, an Error will be thrown.
+     * If the value is already an  List, its own value will be wrapped instead.
+     * Any list elements that are not already  values will be converted automatically.
+     *
+     * @param value
+     */
+    list (value: any[]) :  ListLiteral;
+    /**
+     * Object
+     *
+     * Wraps the given value as an  Object literal.
+     *
+     * qb.obj(value)
+     *
+     * If the value is not a JavaScript Object, an Error will be thrown.
+     * If the value is already an  Object, its own value will be wrapped instead.
+     * Any property values that are not already  values will be converted automatically.
+     * Any keys that are quoted strings will be treated as string literals.
+     * Any keys that start with the character ":" will be treated as dynamic properties and must be well-formed simple references.
+     * Any other keys that need escaping will be quoted if necessary.
+     * If you need to pass in raw JavaScript objects that shouldn't be converted according to these rules, you can use the qb function directly instead.
+     *
+     * Examples
+     *
+     * qb.obj({'some.name': 'value'}) :  {"some.name": value}
+     *
+     * qb.obj({hello: world}) :  {hello: world}
+     *
+     * qb.obj({'"hello"': world}) :  {"hello": world}
+     *
+     * qb.obj({':dynamic': 'props'}) :  {[dynamic]: props}
+     *
+     * qb.obj({': invalid': 'key'}) :  throws an error (invalid is not a well-formed reference)
+     *
+     * @param obj
+     */
+    obj (obj: {}) :  ObjectLiteral;
```

Comment:
If you want to guarantee that the `obj` parameter isn't, eg, `2`, you want to use the `object` type here, and not `{}`. (But only if you bump the Typescript Version in the header to 2.2)